app/controllers/cart_controller.rb

```diff
@@ -1,4 +1,6 @@
 class CartController < ApplicationController
+  before_action :require_not_admin
```

Comment:
This before_action in the CartController is good since all users will be able to access the cart except an admin